android/src/main/java/com/marigold/rnsdk/RNEngageBySailthruModule.kt

```diff
@@ -0,0 +1,316 @@
+            override fun onSuccess() {
+                promise.resolve(true)
+            }
+
+            override fun onFailure(error: Error) {
+                promise.reject(RNMarigoldModule.ERROR_CODE_TRACKING, error.message)
+            }
+        })
+    }
+
+    @ReactMethod
+    fun setProfileVars(vars: ReadableMap, promise: Promise) {
+        val varsJson = try {
+            jsonConverter.convertMapToJson(vars)
+        } catch (e: JSONException) {
+            promise.reject(RNMarigoldModule.ERROR_CODE_VARS, e.message)
+            return
+        }
+        createEngageBySailthru().setProfileVars(varsJson, object : Marigold.MarigoldHandler<Void?> {
+            override fun onSuccess(value: Void?) {
+                promise.resolve(true)
+            }
+
+            override fun onFailure(error: Error) {
+                promise.reject(RNMarigoldModule.ERROR_CODE_VARS, error.message)
+            }
+        })
+    }
+
+    @ReactMethod
+    fun getProfileVars(promise: Promise) {
+        createEngageBySailthru().getProfileVars(object : Marigold.MarigoldHandler<JSONObject?> {
+            override fun onSuccess(value: JSONObject?) {
+                try {
+                    val vars = value?.let { jsonConverter.convertJsonToMap(it) }
+                    promise.resolve(vars)
+                } catch (e: JSONException) {
+                    promise.reject(RNMarigoldModule.ERROR_CODE_VARS, e.message)
+                }
+            }
+
+            override fun onFailure(error: Error) {
+                promise.reject(RNMarigoldModule.ERROR_CODE_VARS, error.message)
+            }
+        })
+    }
+
+    @ReactMethod
+    fun logPurchase(purchaseMap: ReadableMap, promise: Promise) {
+        val purchase = getPurchaseInstance(purchaseMap, promise) ?: return
+        createEngageBySailthru().logPurchase(purchase, object : Marigold.MarigoldHandler<Void?> {
+            override fun onSuccess(value: Void?) {
+                promise.resolve(true)
+            }
+
+            override fun onFailure(error: Error) {
+                promise.reject(RNMarigoldModule.ERROR_CODE_PURCHASE, error.message)
+            }
+        })
+    }
+
+    @ReactMethod
+    fun logAbandonedCart(purchaseMap: ReadableMap, promise: Promise) {
+        val purchase = getPurchaseInstance(purchaseMap, promise) ?: return
+        createEngageBySailthru().logAbandonedCart(purchase, object : Marigold.MarigoldHandler<Void?> {
+            override fun onSuccess(value: Void?) {
+                promise.resolve(true)
+            }
+
+            override fun onFailure(error: Error) {
+                promise.reject(RNMarigoldModule.ERROR_CODE_PURCHASE, error.message)
+            }
+        })
+    }
+
+    override fun getName(): String {
+        return "RNEngageBySailthru"
+    }
+
+    //Helper method for instantiating EngageBySailthru
+    fun createEngageBySailthru(): EngageBySailthru {
```

Comment:
I think this helper method makes sense if we were to try-catch the error, report it and return null. If we're just going to let the Exception be thrown straight to the JS layer we might as well just replace this with `EngageBySailthru()`. Thoughts?